tests/expressions_window/tests.py

```diff
@@ -0,0 +1,755 @@
+import datetime
+from unittest import skipIf, skipUnless
+
+from django.core.exceptions import FieldError
+from django.db import NotSupportedError, connection
+from django.db.models import F, RowRange, ValueRange, Window
+from django.db.models.aggregates import Avg, Max, Min, Sum
+from django.db.models.expressions import ExpressionList, OrderBy, Value
+from django.db.models.functions import (
+    CumeDist, DenseRank, ExtractYear, FirstValue, Lag, LastValue, Lead,
+    NthValue, Ntile, PercentRank, Rank, RowNumber,
+)
+from django.test import TestCase, skipUnlessDBFeature
+
+from .models import WindowTestModel
+
+
+@skipUnlessDBFeature('supports_over_clause')
+class TestWindowFunction(TestCase):
```

Comment:
The test data can most likely be simplified. I was aiming at something that can be partitioned and ordered in different ways. Probably quite useful to have some notes about the expected output :+1: